.github/workflows/unit-tests.yml

```diff
@@ -0,0 +1,97 @@
+name: Unit tests
+
+on:
+  pull_request:
+  push:
+
+jobs:
+  style:
+    name: Style
+    runs-on: ubuntu-latest
+    steps:
+      - name: Clone project
+        uses: actions/checkout@v2
+      - name: hlint
+        run: |
+          set -ex
+          ./etc/scripts/get-hlint.sh
+          export PATH="$(pwd)"/hlint:$PATH
+          hlint src/
+          hlint src/ --cpp-define=WINDOWS=1
+          hlint test/ --cpp-simple
```

Comment:
I guess we could use the one available in marketplace: https://github.com/marketplace/actions/haskell-linter

But it's unofficial (not from github) and we have a better control here. From looking at the repo, I wasn't able to find which hlint version it was using.

Reply:
I guess adding hlint to the official one would add value: https://github.com/github/super-linter

Reply:
No objection to doing that in the future, but I don't want to block this PR on it. I also don't want to rely on something unofficial, especially when the solution we have has worked on Azure CI for years.